app/file-upload/ufs/AmazonS3/server.js

```diff
@@ -137,12 +147,18 @@ export class AmazonS3Store extends UploadFS.Store {
 				}
 			});
 
-			s3.putObject({
+			const params = {
 				Key: this.getPath(file),
 				Body: writeStream,
 				ContentType: file.type,
+			};
+
+			if (classOptions.SSECustomerAlgorithm && classOptions.SSECustomerKey) {
```

Comment:
This if can be removed to a function that validates these variables, and manipulate the params

Reply:
they can be, but it might add unneeded complexity.